core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/ExtensionLoader.java

```diff
@@ -92,8 +93,16 @@ public static void loadVault(ServiceExtensionContext context) {
     }
 
     public static @NotNull Telemetry loadTelemetry() {
-        // TODO: enable overriding telemetry through an extension
-        return new Telemetry(GlobalOpenTelemetry.get());
+        var loader = ServiceLoader.load(OpenTelemetry.class);
+        var openTelemetries = loader.stream().map(ServiceLoader.Provider::get).collect(Collectors.toList());
+        return new Telemetry(loadOpenTelemetry(openTelemetries));
+    }
+
+    public static @NotNull OpenTelemetry loadOpenTelemetry(List<OpenTelemetry> openTelemetries) {
```

Comment:
This method can be package protected as its only visible for testing.

Reply:
Good point! 👍